models/MigrationService.cfc

```diff
@@ -1,23 +1,74 @@
 component singleton accessors="true" {
 
-	property name="wirebox" inject="wirebox";
+	property name="wirebox" 			inject="wirebox";
+	property name="environment" 		inject="coldbox:setting:environment" default="development";
+	property name="manager"             default="cfmigrations.models.QBMigrationManager";
 	property name="migrationsDirectory" default="/resources/database/migrations";
-	property name="datasource";
-	property name="defaultGrammar"  default="AutoDiscover@qb";
-	property name="schema"          default="";
-	property name="migrationsTable" default="cfmigrations";
+	property name="seedsDirectory"      default="/resources/database/seeds";
+	property name="seedEnvironments"    default="development";
+	property name="managerProperties";
+
+
+	MigrationService function init() {
```

Comment:
Please note that by NOT defining the init arguments and just doing them dynamically you loose documentation on the way to initialize this object.  You are assuming the right arguments will be passed and there is no validation for the incoming data.  Why the hesitation of strong typing the arguments?

Reply:
That's fair.  The downside is I have to add a null check for each item in the arguments, when looping the keys.

Reply:
Done.